core/src/mindustry/entities/Lightning.java

```diff
@@ -43,7 +43,9 @@ private static void createLightningInternal(@Nullable Bullet hitter, int seed, T
         bhit = false;
 
         for(int i = 0; i < length / 2; i++){
-            hitCreate.create(null, team, x, y, 0f, damage, 1f, 1f, hitter);
+            if(hitter != null && (hitter.type.lightningTypeChance < Mathf.random() || hitter.type.lightningTypeChance >= 1)){
```

Comment:
`if(hitter != null && Mathf.chance(hitter.type.lightningTypeChance)){`
Use Mathf.chance(), its shorter.

Reply:
Im on mobile and have no idea how to make code suggestions on here so you'll have to do that yourself